### DISCUSSION TITLE
Does this lib uses free of charge Google mobile SDK, or does it implies a fee per map load?

### DISCUSSION THREAD

**Discussion by DoneDeal0** [react-native-maps/react-native-maps#4278] [Q&A]
Hello,

I know this is not a bug, but since you don't have a discussion tab, I didn't know where to send this message. So, I'd love to use your lib in an Expo app, but I need to be sure of its actual cost. Does this lib uses free of charge Google mobile SDK, or does it implies a fee per map load (currently 7$ per 1000 load)?

Long story short, if I have 1000 users displaying 50 maps per month each, will it cost me:
a) 0$ (thanks to free SDK)
b) 150.50$ (50.000 maps - 28.500 free maps =  21.500 maps,  at a 7$ per 1000 load fee)

Thank you!

**Comment by monholm** [comment]
`react-native-maps` uses the Maps SDK for Android and iOS :)

**Comment by DoneDeal0** [comment]
Very nice! Thanks!

**Comment by DoneDeal0** [comment]
Hi,

So I activated the Android SDK in Google Cloud, and Google asked me to give them my credit card number. It seemed weird since your library is supposed to be free. Someone at Expo told me that "the SDK is free, but the api that this SDK is calling is not". 

Basically, if a user displays 25 maps per month (which is a very low number), it means I can only have 1120 free users (“best” case scenario)… In such case, I wouldn't be able to use this map, since I wouldn't even know how much ads I would need to cover the cost of this lib. 

Do you confirm your previous message according to which this lib usage is 100% free?

**Reply by RyanTG** [reply]
> It seemed weird since your library is supposed to be free.

The library is free. The SDK is not the react-native-maps library.

Semantics aside, the maps sdk usage is free. You are pointing out that they ask for a CC#. Well, it’s still free.

**Reply by WillSmithTE** [reply]
I'm still not really clear on this. Google's [pricing page](https://mapsplatform.google.com/pricing/) clearly writes: 
Maps SDK for Android: $7/1000 requests. Then they say `Get mobile Dynamic Maps without Cloud-based maps styling at no cost`, but the [link](https://developers.google.com/maps/billing/gmp-billing?_gl=1*r6q21c*_ga*MTQ5MTUxOTE5NC4xNjU5NDYzMzA5*_ga_NRWSTWS78N*MTY2MDA1MTkwNC4zLjEuMTY2MDA1MjgzMy4w#mobile-dynamic) seemingly mentions nothing about free maps. Finally the FAQ writes `Note that the Maps Embed API, Maps SDK for Android, and Maps SDK for iOS currently have no usage limits and are at no charge (usage of the API or SDKs is not applied against your $200 monthly credit).`, which seems totally at-odds with the $7/1000.

I understand the $200 credit, but there's a huge difference between $7/1000 with the first $200 free, and a totally free SDK. Am I missing something obvious here? I guess this is more of a google question but I would love to hear from others using this lib on their experience

**Reply by RyanTG** [reply]
Yeah, it's confusing and I agree it's at odds. It's also very challenging to make sense of the usage pages. For example, I'm showing 110,763 queries for the iOS SDK and 55,405 for the Android SDK (last 30 days). On the Billing section of the Google Cloud console, it lists "Mobile Native Dynamic Maps" as the SDK (I also use the geocoding API, and seem to average like 1k queries per month). There is no breakdown showing if I've eaten into the $200 credits.

This site - https://bigboomdesign.com/google-maps-api-pricing-and-options/ - notes that this "Mobile Native Dynamic Maps" is $0, and that the $7/1000 applies to other SDKs.... maybe?

Hopefully others can chime in with their experience. I don't think I'm grandfathered into anything special.

**Reply by WillSmithTE** [reply]
Ok thanks a lot, maybe this `Mobile Native Dynamic Maps` is the key, because [this page](https://developers.google.com/maps/documentation/android-sdk/usage-and-billing) also shows it to be $0. I guess the [main pricing page](https://mapsplatform.google.com/pricing/) just doesn't take this into account, which is a shame for Google because I would've been turned off by the $7/1000 graphic if I hadn't found discussions like these.

**Reply by masafumimori** [reply]
Just to be extra clear on the pricing, using Map ID costs depending on the number of requests per month while without Map ID it is for free of charge, I believe?
Based on the description [here](https://developers.google.com/maps/documentation/android-sdk/usage-and-billing)?
Is it correct?

**Reply by relasine** [reply]
We're looking for answers on this at our shop at well. We're looking at something like 200,000 map loads per-month, and there's a big difference between free and $1400.

**Reply by DoneDeal0** [reply]
My app had absolutely no success, so sadly, I still don't know how this SDK handles 100k map loads per month. I think it's free though.

**Reply by RyanTG** [reply]
https://developers.google.com/maps/billing-and-pricing/pricing#mobile-dynamic $0

**Reply by ChristopherMorales** [reply]
Does this still true in 2023? I'm working on a side project and I just want to display a map with some markers so the users know where that place is located. 

Does anyone has experience launching this to production and being charged for API hits with the Android client? Is anyone being charged for just displaying the map?

**Reply by RyanTG** [reply]
Yes, it's still true. There was a clear SO answer that helped explain why the pricing info is so confusing, but unfortunately SO deleted it. But, as I remember, the gist is that dynamic maps SDK usage is free for mobile, _except_ if you use a Map ID and make a custom map style. I have no experience doing that... However, I do use a custom dark mode map style, and it's still free for me! Maybe because I'm not using a Map ID.

In conclusion, still free, except under very rare circumstances.

**Reply by ChristopherMorales** [reply]
Thanks for the response and for clarifying my doubt.

**Comment by ssommelet21** [comment]
Thanks to RyanTG for all the precise explanations 👍
